lib/core/util.js

```diff
@@ -199,7 +199,13 @@ function destroy (stream, err) {
       // See: https://github.com/nodejs/node/pull/38505/files
       stream.socket = null
     }
-    stream.destroy(err)
+
+    // HTTP/2 - It causes to throw uncaught exceptions due to the
+    // error passed down the socket.destroy
+    const isHTTP2 = stream.alpnProtocol === 'h2' &&
+      Object.getPrototypeOf(err).constructor === ClientDestroyedError
```

Comment:
Are we missing an `error` handler?

Reply:
Fixed, good catch. When creating the H2 session, the socket didn't have its event handlers